### ISSUE TITLE
Rule E3004 is not checking for DependsOn declarations

### ISSUE THREAD

**Issue by miparnisari** [aws-cloudformation/cfn-python-lint#1386]
*cfn-lint version*: 0.8.2

*Description of issue.*

The following template has a circular dependency:
```yaml
Resources:
  Resource:
    Type: AWS::SNS::Topic
    DependsOn: Resource2
  Resource2:
    Type: AWS::SNS::Topic
    DependsOn: Resource
```

but `cfn-lint` doesn't report any problems with it.

**Comment by PatMyron** [comment]
Yup, code and description in this file would need to be updated:

https://github.com/aws-cloudformation/cfn-python-lint/blob/35a1fa54e828ce5497683551e9c43b4b210b4227/src/cfnlint/rules/resources/CircularDependency.py#L13-L16

Couple other rules have examples grabbing [`DependsOn`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-dependson.html) attribute values:

https://github.com/aws-cloudformation/cfn-python-lint/blob/b788cc9bd3d49ed20d5f2e58602755a0ef37f52c/src/cfnlint/rules/resources/DependsOn.py#L40

https://github.com/aws-cloudformation/cfn-python-lint/blob/4e210a2325ca225ff8483ec66f0ca9096cd4f73f/src/cfnlint/rules/resources/DependsOnObsolete.py#L64
